tmva/sofie/inc/TMVA/ROperator_Concat.hxx

```diff
@@ -114,8 +114,10 @@
             out<<"\n//--------- Concat\n";
             // special case for 0 axis that memory is contigous
             if (fAxis == 0) {
```

Comment:
I think we can also use the condition of fAxis=0 if  the input shapes have all one's before. 
We can check if the product of the input shape before fAxis is =1 , i.e.
```
bool hasShapeOnes = true;
for (int i = 0; i < fAxis; i++) {
      if (fInputShapes[0][i] != 1) hasShapeOnes = false;
}
if if (fAxis == 0 || hasShapeOnes ) { 
.... // special case that memory is contigous
.....
```

Reply:
Ahh yes, good catch, I will make that change, thanks!